src/main/java/net/wovenmc/woven/api/item/settings/WovenItemSettings.java

```diff
@@ -48,7 +50,7 @@ public WovenItemSettings meter(MeterComponent meterComponent) {
 	 * @param remainder A function for determining the remainder of an item stack when crafting dynamically.
 	 * @return The item settings with the function added.
 	 */
-	public WovenItemSettings dynamicRecipeRemainder(Function<ItemStack, Item> remainder) {
+	public WovenItemSettings dynamicRecipeRemainder(BiFunction<ItemStack, Identifier, ItemStack> remainder) {
```

Comment:
Maybe replace the `BiFunction` with a custom functional interface?
Might be worth it to have a custom interface to have specific documentation on which argument is what.

It's currently not very clear on what `Identifier` is for, and we should not expect users to go read the mixins to know that it corresponds to the recipe identifier.

Reply:
oh right docs

ok ill turn this and the other one into functional interfaces

Reply:
should i do functional interfaces for the meter component too

Reply:
I am not entirely sure, maybe if it improves documentation?

Reply:
added them